GCC dislikes `[[unlikely]]` annotation

In file included from /work/external/iree/iree/base/internal/status.h:23,
--
  | from /work/external/iree/iree/base/status.h:18,
  | from /work/external/iree/iree/base/file_io.h:21,
  | from /work/external/iree/iree/base/file_io_test.cc:15:
  | /work/external/iree/iree/base/internal/status.h: In member function ‘iree::Status& iree::Status::operator=(iree::Status&&)’:
  | /work/external/iree/iree/base/api.h:186:27: error: expected identifier before ‘[’ token
  | 186 \| #define IREE_UNLIKELY(x) [[unlikely]] (x)
  | \|                           ^
  | /work/external/iree/iree/base/internal/status.h:119:11: note: in expansion of macro ‘IREE_UNLIKELY’
  | 119 \|       if (IREE_UNLIKELY(value_)) iree_status_ignore(value_);
  | \|           ^~~~~~~~~~~~~
  | /work/external/iree/iree/base/internal/status.h:119:25: error: ‘value_’ is not a type
  | 119 \|       if (IREE_UNLIKELY(value_)) iree_status_ignore(value_);
  | \|                         ^~~~~~
  | /work/external/iree/iree/base/api.h:186:40: note: in definition of macro ‘IREE_UNLIKELY’
  | 186 \| #define IREE_UNLIKELY(x) [[unlikely]] (x)
  | \|                                        ^

--- a/iree/base/api.h
+++ b/iree/base/api.h
@@ -181,10 +181,7 @@ extern "C" {
 //   if (IREE_UNLIKELY(something_failed)) {
 //     return do_expensive_error_logging();
 //   }
-#if IREE_HAVE_ATTRIBUTE(likely) && IREE_HAVE_ATTRIBUTE(unlikely)
-#define IREE_LIKELY(x) [[likely]] (x)
-#define IREE_UNLIKELY(x) [[unlikely]] (x)
-#elif defined(__GNUC__) || defined(__clang__)
+#if defined(__GNUC__) || defined(__clang__)
 #define IREE_LIKELY(x) (__builtin_expect(!!(x), 1))
 #define IREE_UNLIKELY(x) (__builtin_expect(!!(x), 0))
 #else

